src/go/rpk/pkg/cli/topic/trim.go

```diff
@@ -104,7 +111,10 @@ Trim records from a JSON file
 			}
 			drr, err := adm.DeleteRecords(cmd.Context(), o)
 			out.MaybeDie(err, "unable to trim records: %v", err)
-			printDeleteRecordResponse(drr)
+			ok := printDeleteRecordResponse(drr)
+			if !ok { // This means that at least 1 row contained an error.
+				os.Exit(1)
```

Comment:
not out.Die?

Reply:
This is just to exit with code 1, we won't print any message since it's already in the table created by `printDeleteRecordResponse`